ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -139,17 +140,22 @@
   private final KsqlExecutionContext executionContext;
   private final RoutingFilterFactory routingFilterFactory;
   private final RateLimiter rateLimiter;
+  private final Optional<PullQueryExecutorMetrics> pullQueryMetrics;
+  private final Time time;
 
   public PullQueryExecutor(
       final KsqlExecutionContext executionContext,
       final RoutingFilterFactory routingFilterFactory,
-      final KsqlConfig ksqlConfig
+      final KsqlConfig ksqlConfig,
+      final Optional<PullQueryExecutorMetrics> pullQueryMetrics
   ) {
     this.executionContext = Objects.requireNonNull(executionContext, "executionContext");
     this.routingFilterFactory =
         Objects.requireNonNull(routingFilterFactory, "routingFilterFactory");
     this.rateLimiter = RateLimiter.create(ksqlConfig.getInt(
         KsqlConfig.KSQL_QUERY_PULL_MAX_QPS_CONFIG));
+    this.pullQueryMetrics = pullQueryMetrics;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

Reply:
I now create it here rather than pass it in.